nodejs/webrtc-hello-again/server.js

```diff
@@ -0,0 +1,257 @@
+      }
+  }).catch(error => {
+    console.error('Error in createParticipant:', error);
+    res.send('{"status":"Failed to get token"}')
+  });  
+
+
+
+});
+
+/**
+ * Bandwidth's Voice API will hit this endpoint when an outgoing call is answered
+ */
+app.post("/callAnswered", async (req, res) => {
+  // console.log("Callback from Voice system:", req.body)
+  const callId = req.body.callId;
+  console.log(`received answered callback for call ${callId} tp ${req.body.to}`);
+
+  const participant = calls.get(callId);
+  if (!participant) {
+    console.log(`no participant found for ${callId}!`);
+    res.status(400).send();
+    return;
+  }
+
+  // This is the response payload that we will send back to the Voice API to transfer the call into the WebRTC session
+  // const bxml = webRTCController.generateTransferBxml(participant.token);
+  const bxml = generateTransferBxml(participant.token);
+  console.log(`transferring call ${callId} to session ${sessionId}`)
+
+  // Send the payload back to the Voice API
+  res.contentType("application/xml").send(bxml);
+  console.log("transferred")
+});
+
+/** 
+ * End the Phone Call
+ */
+app.get('/endPSTNCall', async (req, res) => {
+  if (getSessionId() == false || callId == false){
+    console.log("No session found; can't end PSTN call")
+    res.send('{"status":"Failed to end PSTN call; no session id found"}')
+    return;
+  }
+
+  var body = new BandwidthVoice.ApiModifyCallRequest({"state": "completed"});
+  try{
+    var response = await voiceController.modifyCall(accountId, callId, body)
+    res.send('{"status":"hungup"}')
+
+  } catch (e) {
+    console.log(`error hanging up ${process.env.OUTBOUND_PHONE_NUMBER}:`, e);
+    res.send('{"status":"call hangup failed"}')
+  }
+});
+
+
+/**
+ * Support for other files, this isn't really secure for prod deployments
+ * - currently used for the JS SDK
+ */
+app.use(function(req, res, next) {
+  if (!req.route){
+    // undefined path
+    // console.log("unknown path: ", req.path)
+    file  = __dirname + req.path;
+    try {
+      if (fs.existsSync(file)) {
+        //file exists
+        // console.log("serving ", file)
+        res.sendFile(file)
+      } else{
+        // console.log("file not found: ", file)
+        return next ("No file found:", file);
+      }
+    } catch(err) {
+      console.log("Failed to check file")
+      console.error(err)
+    }
+  } 
+  // next();
+});
+/** 
+ * start our server
+ */
+app.listen(port, () => {
+  console.log(`Example app listening on port  http://localhost:${port}`)
+});
+
+/**
+ * @param {sessionId} session_id
+ */
+function saveSessionId (session_id){
+  // saved globally for speed of demo
+  sessionId = session_id;
+}
+function getSessionId(){
+  return sessionId;
+}
+
+/**
+ * Helper method to generate transfer BXML from a WebRTC device token
+ * @param deviceToken device token received from the call control API for a participant
+ */
+function generateTransferBxml (deviceToken){
```

Comment:
Shouldn't the SDK be doing this?

Reply:
There are some issues with the current version of the SDK, so I'm continuing to roll my own. I hope to see resolution of the SDK issue shortly.